### PR TITLE
iOS Search Bar

### PR DESCRIPTION
**Pull request by WillMahler**
- Added iOS search bar, still needs to be connected
   to database

### CODE REVIEW

**Comment by WillMahler** [comment]
This video was the most helpful one I had found when it came to implementing this:
https://www.youtube.com/watch?v=vgvbrBX2FnE

File: ios/atlas/SearchViewModelWrapper.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    init() {
+        searchResults = []
+        viewModel = SearchViewModel(dispatchers: CoroutineDispatchers(), scope: MainScope(), withDatabase: PopulateDatabaseKt.sharedDatabase)
+        
+        viewModel.searchResults.subscribe(
+            onEvent: { result in
+                self.searchResults = result as! Array<SearchResult>
+            },
+            onError: {throwable in
```
Comment (dellisd) [inline ios/atlas/SearchViewModelWrapper.swift:25]:
```suggestion
            onError: { throwable in
```

**Review by dellisd — CHANGES_REQUESTED** [review]

File: ios/atlas/SearchView.swift [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        }
+        .padding(15)
+    }
+}
+
+struct SearchBar: View {
+    
+    @State var query: String = "" {
+        didSet {
+            viewModel.getSearchResults(query: query)
```
[the remark this answers is not in the document]
Comment (safa-howaid) [reply ios/atlas/SearchView.swift:30]: yup

[the remark this answers is not in the document]
Comment (dellisd) [reply ios/atlas/SearchView.swift:30]: Would it make more sense to define it above `query`? 🤔

[the remark this answers is not in the document]
Comment (safa-howaid) [reply ios/atlas/SearchView.swift:30]: probably haha

**Review by dellisd — APPROVED** [review]

**Pull request merged**